applications/StructuralMechanicsApplication/custom_elements/updated_lagrangian.cpp

```diff
@@ -0,0 +1,565 @@
+
+        // Contribution to external forces
+        const Vector body_force = this->GetBodyForce();
+        
+        for ( unsigned int point_number = 0; point_number < integration_points.size(); point_number++ )
+        {
+            // Compute element kinematics B, F, DN_DX ...
+            CalculateKinematicVariables(this_kinematic_variables, point_number, integration_points);
+            
+            // Compute material reponse
+            CalculateConstitutiveVariables(this_kinematic_variables, this_constitutive_variables, Values, point_number, integration_points, GetStressMeasure());
+
+            // Calculating weights for integration on the reference configuration
+            double int_to_reference_weight = GetIntegrationWeight(integration_points, point_number, this_kinematic_variables.detJ0); 
+
+            if ( dimension == 2 && GetProperties().Has( THICKNESS )) 
+            {
+                int_to_reference_weight *= GetProperties()[THICKNESS];
+            }
+
+            if ( CalculateStiffnessMatrixFlag == true ) //calculation of the matrix is required
+            {
+                // Contributions to stiffness matrix calculated on the reference config
+                /* Material stiffness matrix */
+                this->CalculateAndAddKm( rLeftHandSideMatrix, this_kinematic_variables.B, this_constitutive_variables.D, int_to_reference_weight );
+
+                /* Geometric stiffness matrix */
+                this->CalculateAndAddKg( rLeftHandSideMatrix, this_kinematic_variables.DN_DX, this_constitutive_variables.StressVector, int_to_reference_weight );
+            }
+
+            if ( CalculateResidualVectorFlag == true ) //calculation of the matrix is required
+            {
+                this->CalculateAndAddResidualVector(rRightHandSideVector, this_kinematic_variables, rCurrentProcessInfo, body_force, this_constitutive_variables.StressVector, int_to_reference_weight);
+            }
+        }
+
+        KRATOS_CATCH( "" )
+    }
+
+    //************************************************************************************
+    //************************************************************************************
+    
+    void UpdatedLagrangian::CalculateKinematicVariables(
+        KinematicVariables& rThisKinematicVariables,
+        const unsigned int PointNumber,
+        const GeometryType::IntegrationPointsArrayType& IntegrationPoints
+        )
+    {
+        const IntegrationMethod this_integration_method = GetGeometry().GetDefaultIntegrationMethod();
+        
+        // Shape functions
+        rThisKinematicVariables.N = GetGeometry().ShapeFunctionsValues(rThisKinematicVariables.N, IntegrationPoints[PointNumber].Coordinates());
+
+        // Calculating jacobian
+        rThisKinematicVariables.detJ = CalculateDerivativesOnCurrent(rThisKinematicVariables.J, rThisKinematicVariables.InvJ, rThisKinematicVariables.DN_DX, PointNumber, this_integration_method);
```

Comment:
I will change that, besides there is an error in the computation (i mistake the current and previous configuration)